src/kvstore/raftex/RaftPart.cpp

```diff
@@ -1075,9 +1068,20 @@ typename RaftPart::Role RaftPart::processElectionResponses(
                   << proposedTerm;
         term_ = proposedTerm;
         role_ = Role::LEADER;
+        leader_ = addr_;
+        for (auto& host : hosts_) {
+            host->reset();
+        }
+        bgWorkers_->addTask([self = shared_from_this(), proposedTerm] {
+            self->onElected(proposedTerm);
+        });
+        lastMsgAcceptedTime_ = 0;
+        return true;
+    } else {
+        LOG(INFO) << idStr_ << "Didn't get enough votes of term " << proposedTerm;
+        role_ = Role::FOLLOWER;
```

Comment:
by raft protocol, if the votes has been divide by other candidate, maybe no need change the role, when it election timeout, add term and request vote again.

Reply:
Indeed, the problem is that we use proposed term in election, which will make no leader is elected. We'll open another PR to fix it.